tests/rptest/services/admin.py

```diff
@@ -33,10 +59,50 @@ def _url(node, path):
         return f"http://{node.account.hostname}:9644/v1/{path}"
 
     def _request(self, verb, path, node=None, **kwargs):
-        if node is None:
+        if node is None and self._default_node is not None:
+            # We were constructed with an explicit default node: use that one
+            # and do not retry on others.
+            node = self._default_node
+            retry_connection = False
+        elif node is None:
+            # Pick a random node to run this request on.  If that node gives
+            # connection errors we will retry on other nodes.
             node = random.choice(self.redpanda.nodes)
-
-        r = requests.request(verb, self._url(node, path), **kwargs)
+            retry_connection = True
+        else:
+            # We were called with a specific node to run on -- do no retry on
+            # other nodes.
+            retry_connection = False
+
+        if kwargs.get('timeout', None) is None:
+            kwargs['timeout'] = DEFAULT_TIMEOUT
+
+        fallback_nodes = self.redpanda.nodes
+        fallback_nodes = list(
+            filter(lambda n: n.account.hostname != node.account.hostname,
+                   fallback_nodes))
+
+        # On connection errors, retry until we run out of alternative nodes to try
+        # (fall through on first successful request)
+        while True:
+            url = self._url(node, path)
+            self.redpanda.logger.debug(f"Dispatching {verb} {url}")
+            try:
+                r = self._session.request(verb, url, **kwargs)
+            except requests.ConnectionError:
+                if retry_connection and fallback_nodes:
+                    node = random.choice(fallback_nodes)
+                    fallback_nodes = list(
+                        filter(
+                            lambda n: n.account.hostname != node.account.
```

Comment:
I believe that `ClusterNode`'s are directly comparable

Reply:
I was being a bit paranoid here -- ClusterNodes don't have an explicit equality method, so two identical ClusterNodes are only equal if they are the same actual instance (default python equality definition).

But in this context these are all references to the same set of ClusterNode instances, so you're right that this can be made briefer & just directly compare.

Reply:
I was mistaken about equality. When I looked I scrolled to the bottom of the page too quickly and saw the __eq__ method on the cluster object:

https://github.com/confluentinc/ducktape/blob/master/ducktape/cluster/cluster.py#L79

I'm bummed out that it doesn't have the overload, but I also can't think of a place where we are making copies of the object so it should work. Anyway, checking hostname works.